### PR TITLE
executor: speedup unit tests by spliting them into separate suites

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve? 

Speed up unit tests of executor package.

### What is changed and how it works?

- reduce sleep time
- https://github.com/pingcap/check/pull/10 enables us to run tests in different suites parallelly, so split tests in executor package into 4 suites, and each suite contains 60 tests in average. The execution time is reduced from `24s` to `8~10s` on my laptop.
- some tests cannot run parallelly, such as tests using gofail, put them into seqtest directory and run them sequentially;
- some tests call `Exec(select query)` without calling `rs.Close()`, thus leading to goroutine leak, so change them to newly added function `ExecNoRes` which closes result set.

### Check List 

Tests 

N/A

Code changes

N/A

Side effects

N/A

Related changes

N/A

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8618)

### CODE REVIEW

**Comment by zz-jason** [comment]
```
The command "./hack/check-tidy.sh" exited with 1.
```
@eurekaka Please fix CI

**Comment by eurekaka** [comment]
@zz-jason waiting for https://github.com/pingcap/check/pull/10 to be merged first.

**Comment by eurekaka** [comment]
@winoros Hmm, if we find that the newly added test case slows down the UT execution of the package, we need to find out which suite has least tests(actually we should find the suites with lowest execution time, but we don't account for it now) and put this test into it, or we create a new test suite?

**Comment by eurekaka** [comment]
Wait for https://github.com/pingcap/check/pull/11 to be merged first.

File: executor/builder.go [hunk 1]
```diff
@@ -1029,8 +1029,8 @@ func (b *executorBuilder) buildProjBelowAgg(aggFuncs []*aggregation.AggFuncDesc,
 	}
 
 	return &ProjectionExec{
-		baseExecutor:  newBaseExecutor(b.ctx, expression.NewSchema(projSchemaCols...), projFromID, src),
-		numWorkers:    b.ctx.GetSessionVars().ProjectionConcurrency,
+		baseExecutor: newBaseExecutor(b.ctx, expression.NewSchema(projSchemaCols...), projFromID, src),
+		//numWorkers:    b.ctx.GetSessionVars().ProjectionConcurrency,
```
Comment (zz-jason) [inline executor/builder.go]: Some UT failed because of this?

Comment (eurekaka) [reply executor/builder.go]: Yes, this would cause goroutine leak in `TestAggregation`.

Comment (zz-jason) [reply executor/builder.go]: caused by this issue: https://github.com/pingcap/tidb/issues/8787

**Pull request merged**